src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLogger.cs

```diff
@@ -69,6 +70,33 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             _queueProcessor.EnqueueMessage(new LogMessageEntry(computedAnsiString, logAsError: logLevel >= Options.LogToStandardErrorThreshold));
         }
 
+        /// <inheritdoc />
+        public void LogRecords(IEnumerable<BufferedLogRecord> records)
+        {
+            ThrowHelper.ThrowIfNull(records);
+
+            t_stringWriter ??= new StringWriter();
+
+            foreach (var rec in records)
+            {
+                var logEntry = new LogEntry<BufferedLogRecord>(rec.LogLevel, _name, rec.EventId, rec, null, (s, e) => s.FormattedMessage ?? string.Empty);
+                Formatter.Write(in logEntry, null, t_stringWriter);
+
+                var sb = t_stringWriter.GetStringBuilder();
+                if (sb.Length == 0)
+                {
+                    continue;
+                }
+                string computedAnsiString = sb.ToString();
+                sb.Clear();
+                if (sb.Capacity > 1024)
+                {
+                    sb.Capacity = 1024;
+                }
```

Comment:
Why lower the capacity _during_ the foreach. It's good to avoid having each thread hold onto a large StringBuilder for some prolonged period of time, but wouldn't it be better to do this check after the foreach loop rather than inside it?

Reply:
Done. Was  a cut & paste from Log function already in the code, I didn't clean it up after putting this in a loop.